api/src/main/java/io/keploy/service/GrpcService.java

```diff
@@ -155,6 +164,75 @@ public static void put(Service.TestCaseReq testCaseReq) {
         }
     }
 
+    /**
+     * Filter.  This method is used to filter the test cases based on the regex provided in the configuration
+     *
+     * @param testCaseReq the test case req object which contains all the information about the test case
+     * @return the boolean  true if the test case is valid to be recorded, false otherwise
+     */
```

Comment:
I think you have little bit misunderstood the meanings of the accept and reject.
- In case of acceptRegex (both header & URL), if any of the field matches then you have to create that testcase.
- In case of rejectRegex(both header & URL), if any of the field matches then you have to discard that testcase.

Your java docs is conflicting with the logic. Please re-analyze this.

Reply:
thanks @gouravkrosx  for illustration 
just to get it straight this time
I should check if test request is accepted by doing the following:
accepted_url <= ( if there are accept url list then check if it matches any of them)
accepted_header <= ( if there are accept header list then check if it matches any of them)

rejected_url <= ( if there are reject url list =>  then check if it matches any of them)
rejected_header <= ( if there are reject header list => then check if it matches any of them)
then if (accepted_url || accepted_header) && !(rejected_url || rejected_header)  we should record the test case

Reply:
@gouravkrosx could you please verify if I got the point right or not?